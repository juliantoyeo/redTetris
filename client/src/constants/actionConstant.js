export const ACTIONS = {
	ALERT_POP: 'ALERT_POP'
}

export const PLAYER_ACTIONS = {
	CREATE_PLAYER: 'CREATE_PLAYER',
	UPDATE_PLAYER: 'UPDATE_PLAYER'
}

export const ROOM_ACTIONS = {
	ADD_ROOM: 'ADD_ROOM',
	DELETE_ROOM: 'DELETE_ROOM',
	UPDATE_ROOM: 'UPDATE_ROOM'
}