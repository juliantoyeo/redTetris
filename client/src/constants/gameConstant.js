export const BOARD_SIZE = {
	WIDTH: 10,
	HEIGHT: 20
}

export const MAX_PLAYER = 8;

export const CELL_SIZE = 1.5;

export const COLORS = {
	BLACK: '50, 50, 50',
	BLUE: '110,181,255',
	GREEN: '186, 255, 201',
	DARK_GREEN: '100, 200, 100',
	LIGHT_BLUE: '186,225,255',
	ORANGE: '255,223,186',
	PURPLE: '238,203,255',
	RED: '255,179,186',
	DARK_RED: '255,150,150',
	YELLOW: '255,255,186'
}

export const KEY_CODE = {
	A: 65,
	D: 68,
	S: 83,
	W: 87,
	X: 88,
	Z: 90,
	DOWN: 40,
	LEFT: 37,
	RIGHT: 39,
	SPACE: 32,
	UP: 38
}

export const MAX_ROTATION = 3;

export const WALL_KICK = [
	[{x : -1, y : 0}, {x : -1, y : -1}, {x : 0, y : 2}, {x : -1, y : 2}],
	[{x : 1, y : 0}, {x : 1, y : 1}, {x : 0, y : -2}, {x : 1, y : -2}],
	[{x : 1, y : 0}, {x : 1, y : -1}, {x : 0, y : 2}, {x : 1, y : 2}],
	[{x : -1, y : 0}, {x : -1, y : 1}, {x : 0, y : -2}, {x : -1, y : -2}]
]

export const WALL_KICK_I = [
	[{x : -2, y : 0}, {x : 1, y : 0}, {x : -2, y : 1}, {x : 1, y : -2}],
	[{x : -1, y : 0}, {x : 2, y : 0}, {x : -1, y : -2}, {x : 2, y : 1}],
	[{x : 2, y : 0}, {x : -1, y : 0}, {x : 2, y : -1}, {x : -1, y : 2}],
	[{x : 1, y : 0}, {x : -2, y : 0}, {x : 1, y : 2}, {x : -2, y : -1}]
]

export const PIECES = {
	'0': {
		shape: [[['0']]],
		color: '0, 0, 0'
	},
	'X': {
		shape: [[['0']]],
		color: COLORS.BLACK
	},
	I: {
		shape: [
			[
				['0', '0', '0', '0'],
				['I', 'I', 'I', 'I'],
				['0', '0', '0', '0'],
				['0', '0', '0', '0']
			],
			[
				['0', '0', 'I', '0'],
				['0', '0', 'I', '0'],
				['0', '0', 'I', '0'],
				['0', '0', 'I', '0']
			],
			[
				['0', '0', '0', '0'],
				['0', '0', '0', '0'],
				['I', 'I', 'I', 'I'],
				['0', '0', '0', '0']
			],
			[
				['0', 'I', '0', '0'],
				['0', 'I', '0', '0'],
				['0', 'I', '0', '0'],
				['0', 'I', '0', '0']
			]
		],
		color: COLORS.LIGHT_BLUE
	},
	J: {
		shape: [
			[
				['J', '0', '0'],
				['J', 'J', 'J'],
				['0', '0', '0']
			],
			[
				['0', 'J', 'J'],
				['0', 'J', '0'],
				['0', 'J', '0']
			],
			[
				['0', '0', '0'],
				['J', 'J', 'J'],
				['0', '0', 'J']
			],
			[
				['0', 'J', '0'],
				['0', 'J', '0'],
				['J', 'J', '0']
			]
		],
		color: COLORS.BLUE
	},
	L: {
		shape: [
			[
				['0', '0', 'L'],
				['L', 'L', 'L'],
				['0', '0', '0']
			],
			[
				['0', 'L', '0'],
				['0', 'L', '0'],
				['0', 'L', 'L']
			],
			[
				['0', '0', '0'],
				['L', 'L', 'L'],
				['L', '0', '0']
			],
			[
				['L', 'L', '0'],
				['0', 'L', '0'],
				['0', 'L', '0']
			]
		],
		color: COLORS.ORANGE
	},
	O: {
		shape: [
			[
				['O', 'O'],
				['O', 'O']
			],
			[
				['O', 'O'],
				['O', 'O']
			],
			[
				['O', 'O'],
				['O', 'O']
			],
			[
				['O', 'O'],
				['O', 'O']
			]
		],
		color: COLORS.YELLOW
	},
	S: {
		shape: [
			[
				['0', 'S', 'S'],
				['S', 'S', '0'],
				['0', '0', '0']
			],
			[
				['0', 'S', '0'],
				['0', 'S', 'S'],
				['0', '0', 'S']
			],
			[
				['0', '0', '0'],
				['0', 'S', 'S'],
				['S', 'S', '0']
			],
			[
				['S', '0', '0'],
				['S', 'S', '0'],
				['0', 'S', '0']
			]
		],
		color: COLORS.GREEN
	},
	Z: {
		shape: [
			[
				['Z', 'Z', '0'],
				['0', 'Z', 'Z'],
				['0', '0', '0']
			],
			[
				['0', '0', 'Z'],
				['0', 'Z', 'Z'],
				['0', 'Z', '0']
			],
			[
				['0', '0', '0'],
				['Z', 'Z', '0'],
				['0', 'Z', 'Z']
			],
			[
				['0', 'Z', '0'],
				['Z', 'Z', '0'],
				['Z', '0', '0']
			]
		],
		color: COLORS.RED
	},
	T: {
		shape: [
			[
				['0', 'T', '0'],
				['T', 'T', 'T'],
				['0', '0', '0']
			],
			[
				['0', 'T', '0'],
				['0', 'T', 'T'],
				['0', 'T', '0']
			],
			[
				['0', '0', '0'],
				['T', 'T', 'T'],
				['0', 'T', '0']
			],
			[
				['0', 'T', '0'],
				['T', 'T', '0'],
				['0', 'T', '0']
			]
		],
		color: COLORS.PURPLE
	}
}

export const TEST_BOARD = {
	L_SPIN: [
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','J','0','0','0','0','0'],
		['0','0','0','0','J','J','J','0','0','0'],
		['0','0','0','0','0','I','0','0','0','0'],
		['0','0','0','0','0','I','0','J','J','J'],
		['0','0','0','0','0','I','0','Z','Z','J'],
		['0','0','0','0','0','I','0','0','Z','Z']
	],
	J_SPIN: [
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','T','0'],
		['0','0','0','0','0','0','0','T','T','T'],
		['0','0','0','0','0','0','0','0','L','L'],
		['0','0','0','0','0','0','0','0','0','L'],
		['0','0','0','J','J','Z','J','0','0','L'],
		['0','0','0','J','Z','Z','J','0','O','O'],
		['0','0','0','J','Z','J','J','0','O','O']
	],
	S_SPIN: [
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','J','0','0','0','0','0'],
		['0','0','0','0','J','J','J','0','0','0'],
		['0','0','0','0','I','I','0','0','0','0'],
		['0','0','0','0','I','I','L','0','L','L'],
		['0','0','0','0','I','I','L','0','0','L'],
		['0','0','0','0','I','I','L','L','0','L']
	],
	T_SPIN: [
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','I','I','0','0','0','0'],
		['0','0','0','0','I','I','L','0','0','L'],
		['0','0','0','0','I','I','0','0','0','L'],
		['0','0','0','0','I','I','L','0','L','L']
	]
	,
	I_SPIN: [
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['0','0','0','0','0','0','0','0','0','0'],
		['L','L','L','L','L','L','0','L','L','L'],
		['L','L','L','L','L','L','0','L','L','L'],
		['L','L','L','L','L','L','0','L','L','L'],
		['L','L','L','0','0','0','0','L','L','L'],
		['L','L','L','0','L','L','L','L','L','L'],
		['L','L','L','0','L','L','L','L','L','L'],
		['L','L','L','0','L','L','L','L','L','L'],
		['L','L','L','0','L','L','L','L','L','L']
	]
}