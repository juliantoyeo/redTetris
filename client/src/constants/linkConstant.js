export const LINKS = {
	HOME: '/',
	GAME: '/game'
}