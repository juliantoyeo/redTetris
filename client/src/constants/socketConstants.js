export const SOCKET_RES = {
	PLAYER_CREATED: 'PLAYER_CREATED',
	PLAYER_NAME_EXIST: 'PLAYER_NAME_EXIST',
	PLAYER_NOT_EXIST: 'PLAYER_NOT_EXIST',
	ROOM_CREATED: 'ROOM_CREATED',
	ROOM_NAME_EXIST: 'ROOM_NAME_EXIST',
	ROOM_DOESNT_EXIST: 'ROOM_DOESNT_EXIST',
	ROOM_UPDATED: 'ROOM_UPDATED',
	ROOM_DELETED: 'ROOM_DELETED'
}

export const SOCKET_EVENTS = {
	START_GAME: 'START_GAME',
	RESTART_GAME: 'RESTART_GAME',
	CREATE_PLAYER: 'CREATE_PLAYER',
	UPDATE_PLAYER: 'UPDATE_PLAYER',
	FECTH_ALL_ROOM: 'FECTH_ALL_ROOM',
	CREATE_ROOM: 'CREATE_ROOM',
	UPDATE_ROOM: 'UPDATE_ROOM',
	DELETE_ROOM: 'DELETE_ROOM',
	UPDATE_BOARD: 'UPDATE_BOARD',
	GET_NEW_STACK: 'GET_NEW_STACK',
	UPDATE_STACK_INDEX: 'UPDATE_STACK_INDEX',
	ADD_BLOCKING_ROW: 'ADD_BLOCKING_ROW',
	SET_PLAYER_GAME_OVER: 'SET_PLAYER_GAME_OVER',
	GAME_IS_OVER: 'GAME_IS_OVER'
}
