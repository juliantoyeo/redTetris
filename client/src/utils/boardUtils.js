import { BOARD_SIZE } from '../constants/gameConstant'
import _ from 'lodash'

export const createBoard = () => {
	return (
		Array.from(new Array(BOARD_SIZE.HEIGHT), () => 
			new Array(BOARD_SIZE.WIDTH).fill('0')
		)
		// TEST_BOARD.I_SPIN
	)
}

export const checkCollision = (piece, boardWithLandedPiece, { x: moveX, y: moveY }) => {
	const newBoard = _.cloneDeep(boardWithLandedPiece)
	// console.log("boardWithLandedPiece", boardWithLandedPiece)
	// console.log(piece)
	for (let y = 0; y < piece.shape.length; y += 1) {
		for (let x = 0; x < piece.shape[y].length; x += 1) {
			const moveToY = y + piece.pos.y + moveY
			const moveToX = x + piece.pos.x + moveX
			if (piece.shape[y][x] !== '0') {
				// if (!newBoard[moveToY] || !newBoard[moveToY][moveToX] || newBoard[moveToY][moveToX][1] !== 'clear') {
				if (!newBoard[moveToY] || !newBoard[moveToY][moveToX] || newBoard[moveToY][moveToX] !== '0') {
					// console.log(piece.shape)
					// console.log("collision happen on x : ", x, " y : ", y, " moveToX : ", moveToX, " moveToY : ", moveToY)
					// colPos = { x: x, y: y}
					// return { collided: true, colPos: { x: x, y: y}}
					return true
				}
			}
		}
	}
}

export const getLandPosition = (piece, boardWithLandedPiece) => {
	// console.log("boardWithLandedPiece", boardWithLandedPiece)
	// console.log(piece)
	// let moveY = BOARD_SIZE.HEIGHT - 1
	let moveY = piece.pos.y
	if (piece.pos.y <= 0)
		moveY = 0
	// console.log("checked moveY : ", moveY, " y : ", moveY - piece.pos.y)
	while (!checkCollision(piece, boardWithLandedPiece, { x: 0, y: moveY - piece.pos.y}))
	{
		
		// if (moveY === BOARD_SIZE.HEIGHT - 1)
		// 	return { x: piece.pos.x, y: moveY - 1}
		moveY++
		// console.log("checked moveY : ", moveY, " y : ", moveY - piece.pos.y)
	}
	moveY -= 1
	if (moveY < 0 && piece.type !== 'I') moveY = 0
	// const landPos = { x: piece.pos.x, y: moveY}
	// console.log("landPos : ", landPos)
	return { x: piece.pos.x, y: moveY}
}
