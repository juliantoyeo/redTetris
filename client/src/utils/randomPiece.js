import { PIECES } from '../constants/gameConstant';

export const randomPiece = () => {
	const types = 'IJLOSZT';
	const randType = types[Math.floor(Math.random() * types.length)];
	return [PIECES[randType].shape[0], randType];
}
