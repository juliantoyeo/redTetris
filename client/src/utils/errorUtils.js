import { SOCKET_RES } from '../constants/socketConstants'

export const errorAlert = (error) => {
	switch (error) {
		case SOCKET_RES.PLAYER_NAME_EXIST:
			alert('Name exist, please try a different name');
			break;
		case SOCKET_RES.ROOM_NAME_EXIST:
			alert('Room name exist, please try a different name');
			break;
		case SOCKET_RES.ROOM_DOESNT_EXIST:
			alert('Room does not exist, please try again');
			break;
		default:
			return;
	}
}