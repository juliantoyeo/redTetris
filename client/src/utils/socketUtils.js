import { SOCKET_RES } from '../constants/socketConstants';
import { errorAlert } from '../utils/errorUtils';

export const doRoomSocketEvent = (socket, setSelectedRoom, event, data) => {
	if (socket) socket.emit(event, data, (res) => {
		if (res.status === 200) {
			if (res.msg === SOCKET_RES.ROOM_DELETED || !data.isJoinRoom) {
				setSelectedRoom(null);
			} else {
				setSelectedRoom(res.room);
			}
		} else {
			errorAlert(res.msg);
		}
	});
}