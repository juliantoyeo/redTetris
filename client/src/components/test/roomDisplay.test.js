import React from 'react';
import { shallow } from 'enzyme';

import RoomDisplay from '../RoomDisplay';
import { createBoard } from '../../utils/boardUtils';
import { CELL_SIZE, PIECES } from '../../constants/gameConstant';

const fakeClick = jest.fn();

const fakeRoom = {
	players: ['fakePlayer1'],
	maxPlayer: 1,
	name: 'fakeRoom',
	owner: 'fakePlayer1',
	isStarted: false
}

describe('Test Board component', () => {
	it('Should render properly when isLobby', () => {
		const wrapper = shallow(
			<RoomDisplay
				room={fakeRoom}
				isLobby={true}
				isOwner={false}
				onClick={fakeClick}
				onLeave={fakeClick}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
	it('Should render properly when is not lobby', () => {
		const wrapper = shallow(
			<RoomDisplay
				room={fakeRoom}
				isLobby={false}
				isOwner={false}
				onClick={fakeClick}
				onLeave={fakeClick}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
	it('Should render properly when isOwner is true', () => {
		const wrapper = shallow(
			<RoomDisplay
				room={fakeRoom}
				isLobby={true}
				isOwner={true}
				onClick={fakeClick}
				onLeave={fakeClick}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
	it('Should render properly when game is Started', () => {
		fakeRoom.isStarted = true;
		const wrapper = shallow(
			<RoomDisplay
				room={fakeRoom}
				isLobby={false}
				isOwner={true}
				onClick={fakeClick}
				onLeave={fakeClick}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
	it('Should render properly when room is not full', () => {
		fakeRoom.isStarted = false;
		fakeRoom.maxPlayer = 2;
		const wrapper = shallow(
			<RoomDisplay
				room={fakeRoom}
				isLobby={false}
				isOwner={true}
				onClick={fakeClick}
				onLeave={fakeClick}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
	it('Should render properly when there is no room', () => {
		const wrapper = shallow(
			<RoomDisplay
				room={null}
				isLobby={true}
				isOwner={true}
				onClick={fakeClick}
				onLeave={fakeClick}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
});

describe('Test Board component', () => {
	it('Should test onleave', () => {
		const wrapper = shallow(
			<RoomDisplay
				room={fakeRoom}
				isLobby={true}
				isOwner={false}
				onClick={fakeClick}
				onLeave={fakeClick}
			/>
		);
		wrapper.find('.leaveRoom').simulate('click');
		expect(wrapper).toMatchSnapshot();
	});
	it('Should test onStartGame', () => {
		const wrapper = shallow(
			<RoomDisplay
				room={fakeRoom}
				isLobby={true}
				isOwner={true}
				onClick={fakeClick}
				onLeave={fakeClick}
			/>
		);
		wrapper.find('.startGame').simulate('click');
		expect(wrapper).toMatchSnapshot();
	});
	it('Should test onJoinGame', () => {
		const wrapper = shallow(
			<RoomDisplay
				room={fakeRoom}
				isLobby={false}
				isOwner={false}
				onClick={fakeClick}
				onLeave={fakeClick}
			/>
		);
		wrapper.find('.joinRoom').simulate('click');
		expect(wrapper).toMatchSnapshot();
	});
});