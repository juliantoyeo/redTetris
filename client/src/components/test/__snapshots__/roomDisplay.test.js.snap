// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Board component Should render properly when game is Started 1`] = `
<div
  key="fakeRoom#"
  style={
    Object {
      "alignItems": "center",
      "border": "1px solid #333",
      "borderRadius": "3vw",
      "display": "flex",
      "height": "5vw",
      "justifyContent": "space-around",
      "margin": "1vw 0",
    }
  }
>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Name : 
    fakeRoom
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Owner : 
    fakePlayer1
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Player Count : 
    1/1
  </span>
  <Button
    className="joinRoom"
    isDisable={true}
    onClick={[Function]}
    style={
      Object {
        "width": "15%",
      }
    }
    text="Game Started"
    type="button"
  />
</div>
`;

exports[`Test Board component Should render properly when is not lobby 1`] = `
<div
  key="fakeRoom#"
  style={
    Object {
      "alignItems": "center",
      "border": "1px solid #333",
      "borderRadius": "3vw",
      "display": "flex",
      "height": "5vw",
      "justifyContent": "space-around",
      "margin": "1vw 0",
    }
  }
>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Name : 
    fakeRoom
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Owner : 
    fakePlayer1
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Player Count : 
    1/1
  </span>
  <Button
    className="joinRoom"
    isDisable={true}
    onClick={[Function]}
    style={
      Object {
        "width": "15%",
      }
    }
    text="Full"
    type="button"
  />
</div>
`;

exports[`Test Board component Should render properly when isLobby 1`] = `
<div
  key="fakeRoom#"
  style={
    Object {
      "alignItems": "center",
      "border": "1px solid #333",
      "borderRadius": "3vw",
      "display": "flex",
      "height": "5vw",
      "justifyContent": "space-around",
      "margin": "1vw 0",
    }
  }
>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Name : 
    fakeRoom
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Owner : 
    fakePlayer1
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Player Count : 
    1/1
  </span>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-evenly",
        "width": "20%",
      }
    }
  >
    <Button
      className="leaveRoom"
      onClick={[Function]}
      style={
        Object {
          "width": "40%",
        }
      }
      text="Leave"
      type="button"
    />
  </div>
</div>
`;

exports[`Test Board component Should render properly when isOwner is true 1`] = `
<div
  key="fakeRoom#"
  style={
    Object {
      "alignItems": "center",
      "border": "1px solid #333",
      "borderRadius": "3vw",
      "display": "flex",
      "height": "5vw",
      "justifyContent": "space-around",
      "margin": "1vw 0",
    }
  }
>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Name : 
    fakeRoom
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Owner : 
    fakePlayer1
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Player Count : 
    1/1
  </span>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-evenly",
        "width": "20%",
      }
    }
  >
    <Button
      className="leaveRoom"
      onClick={[Function]}
      style={
        Object {
          "width": "40%",
        }
      }
      text="Leave"
      type="button"
    />
    <Button
      className="startGame"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(100, 200, 100, 1)",
          "width": "40%",
        }
      }
      text="Start"
      type="button"
    />
  </div>
</div>
`;

exports[`Test Board component Should render properly when room is not full 1`] = `
<div
  key="fakeRoom#"
  style={
    Object {
      "alignItems": "center",
      "border": "1px solid #333",
      "borderRadius": "3vw",
      "display": "flex",
      "height": "5vw",
      "justifyContent": "space-around",
      "margin": "1vw 0",
    }
  }
>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Name : 
    fakeRoom
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Owner : 
    fakePlayer1
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Player Count : 
    1/2
  </span>
  <Button
    className="joinRoom"
    isDisable={false}
    onClick={[Function]}
    style={
      Object {
        "width": "15%",
      }
    }
    text="Join"
    type="button"
  />
</div>
`;

exports[`Test Board component Should render properly when there is no room 1`] = `
<div
  key="empty"
  style={
    Object {
      "alignItems": "center",
      "border": "1px solid #333",
      "borderRadius": "3vw",
      "display": "flex",
      "height": "5vw",
      "justifyContent": "space-around",
      "margin": "1vw 0",
    }
  }
>
  <span>
    No room to join
  </span>
</div>
`;

exports[`Test Board component Should test onJoinGame 1`] = `
<div
  key="fakeRoom#"
  style={
    Object {
      "alignItems": "center",
      "border": "1px solid #333",
      "borderRadius": "3vw",
      "display": "flex",
      "height": "5vw",
      "justifyContent": "space-around",
      "margin": "1vw 0",
    }
  }
>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Name : 
    fakeRoom
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Owner : 
    fakePlayer1
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Player Count : 
    1/2
  </span>
  <Button
    className="joinRoom"
    isDisable={false}
    onClick={[Function]}
    style={
      Object {
        "width": "15%",
      }
    }
    text="Join"
    type="button"
  />
</div>
`;

exports[`Test Board component Should test onStartGame 1`] = `
<div
  key="fakeRoom#"
  style={
    Object {
      "alignItems": "center",
      "border": "1px solid #333",
      "borderRadius": "3vw",
      "display": "flex",
      "height": "5vw",
      "justifyContent": "space-around",
      "margin": "1vw 0",
    }
  }
>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Name : 
    fakeRoom
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Owner : 
    fakePlayer1
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Player Count : 
    1/2
  </span>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-evenly",
        "width": "20%",
      }
    }
  >
    <Button
      className="leaveRoom"
      onClick={[Function]}
      style={
        Object {
          "width": "40%",
        }
      }
      text="Leave"
      type="button"
    />
    <Button
      className="startGame"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(100, 200, 100, 1)",
          "width": "40%",
        }
      }
      text="Start"
      type="button"
    />
  </div>
</div>
`;

exports[`Test Board component Should test onleave 1`] = `
<div
  key="fakeRoom#"
  style={
    Object {
      "alignItems": "center",
      "border": "1px solid #333",
      "borderRadius": "3vw",
      "display": "flex",
      "height": "5vw",
      "justifyContent": "space-around",
      "margin": "1vw 0",
    }
  }
>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Name : 
    fakeRoom
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Room Owner : 
    fakePlayer1
  </span>
  <span
    style={
      Object {
        "overflow": "hidden",
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "20%",
      }
    }
  >
    Player Count : 
    1/2
  </span>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-evenly",
        "width": "20%",
      }
    }
  >
    <Button
      className="leaveRoom"
      onClick={[Function]}
      style={
        Object {
          "width": "40%",
        }
      }
      text="Leave"
      type="button"
    />
  </div>
</div>
`;
